som.py

```diff
@@ -72,7 +73,7 @@ def winner(self, vector):
         :return: indices of winning neuron
         """
         indx = np.argmin(np.sum((self.map - vector) ** 2, axis=2))
-        return np.array([indx / self.x, indx % self.y])
+        return np.array([indx // self.y, indx % self.y])
```

Comment:
using different x and y will cause **IndexError**. Also _indx / self.y_ will return float, so use floor division instead.